script.js

```diff
@@ -1,4 +1,577 @@
+let gameMode = "";
+let diceCount = 0;
+let currentPlayerDiceNumbers = [];
+let currentPlayer = 1;
+let player1FinalNumEachRound = [];
+let player2FinalNumEachRound = [];
+
+const rollOneDice = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+const rollMultipleDice = function (diceCount) {
+  let rollStateMessage = `Player ${currentPlayer}'s turn: <br>`;
+  for (i = 0; i < diceCount; i += 1) {
+    currentPlayerDiceNumbers.push(rollOneDice());
+    let diceId = i + 1;
+    rollStateMessage += `You rolled ${currentPlayerDiceNumbers[i]} for Dice ${diceId}. <br>`;
+  }
+  if (gameMode === HIGHEST_COMBINED_NUMBER_MODE) {
+    rollStateMessage += `Computer is about to automatically concatenate the ${diceCount} digits rolled to create the <b> largest </b> possible number. Please hit the "Submit" button to continue.`;
+  } else if (gameMode === LOWEST_COMBINED_NUMBER_MODE) {
+    rollStateMessage += `Computer is about to automatically concatenate the ${diceCount} digits rolled to create the <b> smallest </b> possible number. Please hit the "Submit" button to continue.`;
+  }
+  return rollStateMessage;
+};
+
+const autoGenerateOptimalCombinedNumber = function () {
+  // MUST initialize this as an empty string or else the value will be undefined during reassignment later
+  let currentPlayerFinalNum = "";
+
+  // Arrange digits from big to small in the array, convert each of them to string and keep them in currentPlayerFinalNum variable
+  if (gameMode === HIGHEST_COMBINED_NUMBER_MODE) {
+    currentPlayerDiceNumbers.sort((a, b) => b - a);
+    for (i = 0; i < currentPlayerDiceNumbers.length; i += 1) {
+      currentPlayerFinalNum += String(currentPlayerDiceNumbers[i]);
+    }
+  }
+  // Arrange digits from small to big in the array, etc..
+  else if (gameMode === LOWEST_COMBINED_NUMBER_MODE) {
+    currentPlayerDiceNumbers.sort((a, b) => a - b);
+    for (i = 0; i < currentPlayerDiceNumbers.length; i += 1) {
+      currentPlayerFinalNum += String(currentPlayerDiceNumbers[i]);
+    }
+  }
+
+  // Transfer the final num to another variable for each player separately for LEADERBOARD later
+  if (currentPlayer === 1) {
+    player1FinalNumEachRound.push(Number(currentPlayerFinalNum));
+  } else if (currentPlayer === 2) {
+    player2FinalNumEachRound.push(Number(currentPlayerFinalNum));
+  }
+  return `Player ${currentPlayer}, your number is ${currentPlayerFinalNum}. <br>
+          Player 2's turn now! <br>
+          Please hit the "Submit" button to continue.`;
+};
+
+const showLeaderboard = function () {
+  // Calculate the sum of all player 1 optimal combined numbers
+  let player1Sum = 0;
+  for (i = 0; i < player1FinalNumEachRound.length; i += 1) {
+    player1Sum += player1FinalNumEachRound[i];
+  }
+
+  // Calculate the sum of all player 2 optimal combined numbers
+  let player2Sum = 0;
+  for (i = 0; i < player2FinalNumEachRound.length; i += 1) {
+    player2Sum += player2FinalNumEachRound[i];
+  }
```

Comment:
This can be abstracted into a generic function:
```
var calculatePlayerSum = function(playerFinalNumEachRound){
    let playerSum = 0
    for (let i = 0; i < playerFinalNumEachRound.length; i += 1){
        playerSum += playerFinalNumEachRound[i];
}

let player1Sum = calculatePlayerSum(player1FinalNumEachRound)
let player2Sum = calculatePlayerSum(player2FinalNumEachRound)
```